tests/FSharp.Compiler.ComponentTests/Conformance/BasicGrammarElements/CustomAttributes/CallerArgumentExpression/CallerArgumentExpression.fs

```diff
@@ -0,0 +1,384 @@
+open System.Runtime.InteropServices
+type A() =
+  static member aa (
+    a,
+    [<CallerMemberName; Optional; DefaultParameterValue "no value">]b: string, 
+    [<CallerLineNumber; Optional; DefaultParameterValue 0>]c: int, 
+    [<CallerArgumentExpressionAttribute("a"); Optional; DefaultParameterValue "no value">]e: string) = 
+    a,b,c,e
+
+let stringABC = "abc"
+assertEqual (A.aa(stringABC)) ("abc", ".cctor", 13, "stringABC")
+assertEqual (A.aa(a = stringABC)) ("abc", ".cctor", 14, "stringABC")
+        """
+        |> withLangVersionPreview
+        |> asExe
+        |> compileAndRun
+        |> shouldSucceed
+        |> ignore
+
+    [<FactForNETCOREAPP>]
+    let ``Can define methods using CallerArgumentExpression with F#-style optional arguments`` () =
+        FSharp """let assertEqual a b = if a <> b then failwithf "not equal: %A and %A" a b
+open System.Runtime.CompilerServices
+open System.Runtime.InteropServices
+type A() =
+  static member aa (
+    a,
+    [<CallerMemberName>] ?b: string, 
+    [<CallerLineNumber>] ?c: int, 
+    [<CallerArgumentExpressionAttribute("a")>] ?e: string) = 
+    let b = defaultArg b "no value"
+    let c = defaultArg c 0
+    let e = defaultArg e "no value"
+    a,b,c,e
+
+let stringABC = "abc"
+assertEqual (A.aa(stringABC)) ("abc", ".cctor", 16, "stringABC")
+assertEqual (A.aa(a = stringABC)) ("abc", ".cctor", 17, "stringABC")
+        """
+        |> withLangVersionPreview
+        |> asExe
+        |> compileAndRun
+        |> shouldSucceed
+        |> ignore
+
+    [<FactForNETCOREAPP(Skip = "Currently cannot get the original text range with #line")>]
+    let ``Can define in F# - with #line`` () =
+        FSharp """let assertEqual a b = if a <> b then failwithf "not equal: %A and %A" a b
+open System.Runtime.CompilerServices
+open System.Runtime.InteropServices
+
+let path = System.IO.Path.Combine(__SOURCE_DIRECTORY__, "test.fs")
+
+# 1 "test.fs"
+type A() =
+  static member aa (
+    a,
+    [<CallerMemberName; Optional; DefaultParameterValue "no value">]b: string, 
+    [<CallerLineNumber; Optional; DefaultParameterValue 0>]c: int, 
+    [<CallerFilePath; Optional; DefaultParameterValue "no value">]d: string, 
+    [<CallerArgumentExpressionAttribute("a"); Optional; DefaultParameterValue "no value">]e: string) = 
+    a,b,c,d,e
+    
+  static member B (``ab c``, [<CallerArgumentExpression "ab c">]?n) =
+    defaultArg n "no value"
+
+let stringABC = "abc"
+assertEqual (A.aa(stringABC)) ("abc", ".cctor", 11, path, "stringABC")
+# 1 "test.fs"
+assertEqual (A.aa(stringABC : string)) ("abc", ".cctor", 1, path, "stringABC : string")
+# 1 "test.fs"
+assertEqual (A.aa(a = (stringABC : string))) ("abc", ".cctor", 1, path, "(stringABC : string)")
+
+
+A.B("abc"
+#line 1
+: string)
+|> assertEqual "\"abc\"
+#line 1
+: string"
+
+
+A.B((+) 1
+#line 1
+        123)
+|> assertEqual "(+) 1
+#line 1
+        123"
+
+
+A.B(#line 1
+  (+) 1
+        123)
+|> assertEqual "(+) 1
```

Comment:
For future maintainers, I think this test setup needs more explanation of the context.